examples/linear_model/plot_sparse_logistic_regression_mnist.py

```diff
@@ -0,0 +1,71 @@
+"""
+=====================================================
+MNIST classfification using multinomial logistic + L1
+=====================================================
+
+Performance of multinomial logistic regression with
+L1 penalty. We use the SAGA algorithm for this purpose, which is fast. Test
+accuracy reaches > 0.8, while weight vectors remains *sparse* and
+*interpretable*.
+"""
+import time
+import matplotlib.pyplot as plt
+import numpy as np
+
+from sklearn.datasets import fetch_mldata
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler
+from sklearn.utils import check_random_state
+
+print(__doc__)
+
+# Author: Arthur Mensch <arthur.mensch@m4x.org>
+# License: BSD 3 clause
+
+# Turn down for faster convergence
+t0 = time.time()
+train_samples = 5000
+
+mnist = fetch_mldata('MNIST original')
+X = mnist.data.astype('float64')
+y = mnist.target
+random_state = check_random_state(0)
+permutation = random_state.permutation(X.shape[0])
+X = X[permutation]
+y = y[permutation]
+X = X.reshape((X.shape[0], -1))
+X = StandardScaler().fit_transform(X)
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=10000)
+X_train = X_train[:train_samples]
+y_train = y_train[:train_samples]
+
+# Turn up tolerance for faster convergence
+clf = LogisticRegression(C=50 / train_samples,
+                         multi_class='multinomial',
+                         penalty='l1', solver='saga', tol=0.1)
+clf.fit(X_train, y_train)
+sparsity = np.mean(clf.coef_ == 0) * 100
+score = clf.score(X_test, y_test)
+# print('Best C % .4f' % clf.C_)
+print("Sparsity with L1 penalty: %.2f%%" % sparsity)
+print("Test score with L1 penalty: %.4f" % score)
+
+# Rescale classification vector for displaying
+coef = clf.coef_.copy()
+coef -= np.min(coef, axis=1)[:, np.newaxis]
+coef /= np.max(coef, axis=1)[:, np.newaxis]
+
+for i in range(1, 10):
+    l1_plot = plt.subplot(3, 3, i)
+    l1_plot.imshow(np.abs(coef[i].reshape(28, 28)), interpolation='nearest',
+                   cmap='binary', vmin=0, vmax=1)
```

Comment:
Please use the red blue colormap of matplotlib such that 0 is white. This should better highlight the sparsity of the filters.